### PR TITLE
Replace poster with requests-toolbelt.

### PR DESCRIPTION
**Pull request by jmchilton**
poster is not Python 3 compatible but requests-toolbelt is - so this is providing yet one more step forward toward Python 3 support (#97).

requests-toolbelt is tested with requests >= 2.1.0 so I also updated that requirement in setup.py.

Additionally, f0a090e includes additional testing to verify new upload technique actual uploads the correct content.

### CODE REVIEW

File: bioblend/galaxyclient.py [hunk 1]
```diff
@@ -7,16 +7,32 @@
...
+    def attach_file(self, path, name=None):
+        """
+        Attach a path to a request payload object.
+
+        :type path: string
+        :param path: Path to file to attach to payload.
+
+        :type name: string
+        :param name: Name to give file, if different than actual pathname.
+        """
```
Comment (nsoranzo) [inline bioblend/galaxyclient.py]: Missing `:rtype:` and `:return:`.

File: bioblend/galaxyclient.py [hunk 2]
```diff
@@ -90,21 +106,23 @@ def make_post_request(self, url, payload, params=None, files_attached=False):
...
             payload = json.dumps(payload)
-            r = requests.post(url, data=payload, headers=self.json_headers,
-                              verify=self.verify, params=params)
-            if r.status_code == 200:
-                return r.json()
-            # @see self.body for HTTP response body
-            raise ConnectionError("Unexpected response from galaxy: %s" %
-                                  r.status_code, body=r.text)
+            headers = self.json_headers
+            params = params
```
Comment (nsoranzo) [inline bioblend/galaxyclient.py]: This is a no-op, can be removed.

File: bioblend/galaxyclient.py [hunk 3]
```diff
@@ -7,16 +7,32 @@
...
-from six.moves.urllib.request import Request, urlopen
+from requests_toolbelt import MultipartEncoder
 import six
 
 from .galaxy.client import ConnectionError
 
 
 class GalaxyClient(object):
+
+    def attach_file(self, path, name=None):
```
Comment (nsoranzo) [inline bioblend/galaxyclient.py]: Not sure this function belongs to this class (does not use self), it may probably be moved to `bioblend/__init__.py`.

**Comment by nsoranzo** [comment]
Awesome, I was too scared to touch this code!

Thanks a lot!!

File: bioblend/galaxyclient.py (hunk 2 above)
Comment (jmchilton) [inline bioblend/galaxyclient.py]: Disagree that it should be removed but I have refactored it in ef2e903 as a compromise.

File: bioblend/galaxyclient.py (hunk 3 above)
Comment (jmchilton) [inline bioblend/galaxyclient.py]: Moved in 8f155a4.

File: bioblend/galaxyclient.py (hunk 1 above)
Comment (jmchilton) [inline bioblend/galaxyclient.py]: Added in 8f155a4.

**Comment by jmchilton** [comment]
Great - thanks for the detailed review and merge @nsoranzo.

**Pull request merged**